src/Parsers/FunctionParameterValuesVisitor.cpp

```diff
@@ -0,0 +1,75 @@
+#include <Parsers/FunctionParameterValuesVisitor.h>
+#include <Parsers/ASTExpressionList.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/parseQuery.h>
+#include <Common/FieldVisitorToString.h>
+#include <Parsers/ASTHelpers.h>
+#include <Common/assert_cast.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+class FunctionParameterValuesVisitor
+{
+public:
+    explicit FunctionParameterValuesVisitor(NameToNameMap & parameter_values_)
+        : parameter_values(parameter_values_)
+    {
+    }
+
+    void visit(const ASTPtr & ast)
+    {
+        if (const auto * expression = ast->as<ASTExpressionList>())
+            visitExpressionList(*expression);
+        for (const auto & child : ast->children)
+            visit(child);
+    }
+
+private:
+    NameToNameMap & parameter_values;
+
+    void visitExpressionList(const ASTExpressionList & expression_list)
+    {
+        if (expression_list.children.size() == 2)
+        {
```

Comment:
let's do early return from function here to avoid too many nested scopes as it will be easier to read.

Reply:
addressed it.